test/e2e/operator/deployment_api.go

```diff
@@ -117,6 +125,8 @@ var _ = deploy.DescribeForSome("API", func(d *deploy.Deployment) bool {
 
 	AfterEach(func() {
 		evWatcher.Stop()
+		evMutex.Lock()
+		defer evMutex.Unlock()
```

Comment:
Consider the following execution sequence:
- AfterEach is reached, locks mutex and deletes the map, then returns, which unlocks the mutex.
- The goroutine handles an event, creates a goroutine for it, which then assigns to a nil map.

Have you considered my proposal in the issue to wait for the goroutine termination before deleting the map?

Reply:
> Consider the following execution sequence:
> 
> * AfterEach is reached, locks mutex and deletes the map, then returns, which unlocks the mutex.
> * The goroutine handles an event, creates a goroutine for it, which then assigns to a nil map.
> 
Yes, I agree. there case still possible.

> Have you considered my proposal in the issue to wait for the goroutine termination before deleting the map?

Pushed new changes with  this.